scripts.py

```diff
@@ -2276,6 +2299,39 @@ def run(self):
                 "Here's your problem: your presentation-ready works are not making it into the materialized view.\n"
             )
             return
+
+        # Check if the works have delivery mechanisms.
+        LPDM = LicensePoolDeliveryMechanism
+        mv_works = mv_works.filter(
+            exists().where(
+                and_(MaterializedWork.data_source_id==LPDM.data_source_id,
+                     MaterializedWork.primary_identifier_id==LPDM.identifier_id)
+            )
+        )
+        if mv_works.count() == 0:
+            self.output.write(
+                "Here's your problem: your works don't have delivery mechanisms.\n"
+            )
+            return
+
+        # Check if the license pools are suppressed.
+        mv_works = mv_works.join(LicensePool).filter(
+            LicensePool.suppressed==False)
+        if mv_works.count() == 0:
+            self.output.write(
+                "Here's your problem: your works' license pools are suppressed.\n"
+            )
+            return
+
+        # Check if the pools have available licenses.
+        mv_works = mv_works.filter(
+            or_(LicensePool.licenses_owned > 0, LicensePool.open_access)
+        )
+        if mv_works.count() == 0:
+            self.output.write(
+                "Here's your problem: your works aren't open access and have no licenses owned.\n"
+            )
+            return
```

Comment:
None of these problems can be fixed automatically, but it's still helpful to know what the problem is.